Feature/action queue system

KacperObara opened the pull request:
Dodano system kolejkowania rozkazów, oraz ich wyświetlanie z możliwością anulowania.

Nergal20 committed:
Synchronization

Nergal20 committed:
Synchronization

Nergal20 committed:
Merge branch 'feature/ActionQueueSystem' of https://github.com/Sweaty-Bacon-Ducks/DigThemGraves into feature/ActionQueueSystem

Nergal20 committed:
Working action queue

Nergal20 committed:
Update ActionQueue.cs

KacperObara commented on Assets/Scripts/Actions/ActionQueue.cs:
```diff
@@ -0,0 +1,61 @@
+            else if (currentAction.IsFinished)
+            {
+                RemoveAction(currentAction);
+                currentAction = null;
```
Zostawiłem tak mimo ostrzeżeń, by nie przypisywać do null. Inaczej pojawiają się problemy z wykonywaniem akcji. Poza tym, wydaje mi się, że w tym przypadku pasuje, gdyż jak nie ma akcji, to powinno być puste (null).

copy-on-meow commented on Assets/Scripts/Actions/ActionDisplay.cs:
```diff
@@ -0,0 +1,27 @@
+    /// <summary>
+    /// Class represents clickable icon in action queue
+    /// </summary>
+    public class ActionDisplay : MonoBehaviour
```
`ActionView` jest lepszą nazwą

Nergal20 committed:
Class name changed

copy-on-meow commented:
Proszę o rozwiązanie #15

Nergal20 committed:
Patches

OnMouseDown changed to OnMouseUp, Added Cancel function to IAction and created class diagram

copy-on-meow commented:
ref. #11

copy-on-meow reviewed (changes requested):
Blocked by #15

Nergal20 committed:
Fixes

copy-on-meow commented on Assets/Scripts/Actions/ActionView.cs:
```diff
@@ -0,0 +1,27 @@
+
+        public void OnClicked()
+        {
+            ActionQueue actionQueue = GameObject.FindGameObjectWithTag("GameManager").GetComponent<ActionQueue>();
```
Wolałbym to jako pole w tej klasie a nie szukanie po tagach. Poza tym spoko

copy-on-meow commented on Assets/Scripts/BuildSlot.cs:
```diff
@@ -9,6 +9,6 @@ public abstract class BuildSlot : MonoBehaviour, IBuildSlot
         public abstract List<Buildable> PossibleBuildables { get; }
 
-        public abstract void OnMouseDown();
+        public abstract void OnMouseUp();
```
Chyba nie musisz tego pisać, bo to jest jako callback w MonoBehaviour, więc wszystkie klasy dziedziczące po `BuildSlot` będą to miały możliwość przesłonięcia tej metody

copy-on-meow commented on Assets/Scripts/Core/IAction.cs:
```diff
@@ -4,7 +4,10 @@ namespace DigThemGraves
 {
     public interface IAction
     {
+        Sprite Sprite { get; set; }
```
A to przypadkiem nie jest elementem widoku poszczególnej akcji? Wydaje mi się że to należy wywalić do klasy zajmującej się widokiem

copy-on-meow commented on Assets/Scripts/ScriptablePlayerAction.cs:
```diff
@@ -5,7 +5,14 @@ namespace DigThemGraves
+
+        public abstract Sprite Sprite { get; set; }
+
+        public void Cancel()
```
Wydaje mi się że zapomniałeś o `abstract`

Nergal20 committed:
Fixes

Nergal20 committed:
Quick fix
